full.async/src/full/async.clj

```diff
@@ -178,3 +205,77 @@
           (>! result acc)
           (async/close! result))))
     result))
+
+;; taken from clojure/core ~ 1.7
```

Comment:
This was just to show that the macro code is coming from core.clj originally. I have had to change it quite a bit anyway.